File: 351001/Boukhvalova/module-publisher/src/main/java/by/boukhvalova/distcomp/dto/TweetResponseTo.java
package by.boukhvalova.distcomp.dto;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.sql.Timestamp;
import java.util.List;

@Data
public class TweetResponseTo {
    long id;
    @JsonProperty("userId")
    @JsonAlias("creatorId")
    long userId;
    @Size(min = 2, max = 64)
    String title;
    @Size(min = 4, max = 2048)
    String content;
    Timestamp created;
    Timestamp modified;
    @JsonProperty("stickers")
    @JsonAlias({"markers", "tags"})
    List<String> stickers;
}

File: 351003/Gorodko/app/Publisher/Dto/TweetResponseTo.cs
using System.Text.Json.Serialization;

namespace Publisher.Dto {
    public class TweetResponseTo : BaseResponseTo {
        [JsonPropertyName("editorId")]
        public long EditorId { get; set; }

        [JsonPropertyName("title")]
        public string Title { get; set; } = string.Empty;

        [JsonPropertyName("content")]
        public string Content { get; set; } = string.Empty;

        [JsonPropertyName("created")]
        public DateTime Created { get; set; }

        [JsonPropertyName("modified")]
        public DateTime Modified { get; set; }

        [JsonPropertyName("stickers")]
        public List<string>? Stickers { get; set; }
    }
}